MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionPresenter.cpp

```diff
@@ -1084,7 +1099,31 @@ void EnggDiffractionPresenter::doFocusing(const EnggDiffCalibSettings &cs,
     alg->execute();
 
     const bool plotFocusedWS = m_view->focusedOutWorkspace();
-    if (plotFocusedWS == true) {
+    int plotType = m_view->currentPlotType();
+    g_log.debug() << "...Plot Type return: " << plotType << std::endl;
+
+    // Qt comboBox returns absurd number for index of 0
+    if (plotType > 3) {
+      plotType = 1;
+    } else {
+      plotType += 1;
+    }
+
+    if (plotFocusedWS == true && 1 == plotType) {
+      if (outWSName == "engggui_focusing_output_ws_bank_1")
+        m_view->plotFocusedSpectrum(outWSName);
+      if (outWSName == "engggui_focusing_output_ws_bank_2")
+        m_view->plotReplacingWindow(outWSName);
+    }
+
+    else if (plotFocusedWS == true && 2 == plotType) {
+      if (outWSName == "engggui_focusing_output_ws_bank_1")
+        m_view->plotFocusedSpectrum(outWSName);
+      if (outWSName == "engggui_focusing_output_ws_bank_2")
+        m_view->plotWaterfallSpectrum(outWSName);
+    }
+
+    else if (plotFocusedWS == true && 3 == plotType) {
```

Comment:
To keep all the methods short, this logic about plotFocussedWS could be moved into a method of its own.